management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java

```diff
@@ -0,0 +1,140 @@
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(create("ws://localhost:8555/echo"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("http://localhost:8001")
+                            .done()
+                        .done()
+                        .service()
+                            .property("connector.server.address", "jmx://localhost:2010/")
+                            .type("management.jmx")
+                            .authorization()
+                                .requireRole(ADMIN)
+                            .done()
+                            .realmName("jmxrealm")
+                        .done()
+                        .security()
+                            .trustStore(trustStore)
+                            .keyStore(keyStore)
+                            .keyStorePassword(password)
+                            .keyStoreFile(TlsTestUtil.getKeystoreFileLocation())
+                            .realm()
+                                .name("jmxrealm")
+                                .description("realm for jmx")
+                                .httpChallengeScheme("Application Basic")
+                                .loginModule()
+                                    .type("file")
+                                    .success("required")
+                                    .option("file", "jaas-config.xml")
+                                .done()
+                            .done()
+                        .done()
+                        .property(GATEWAY_CONFIG_DIRECTORY_PROPERTY, "/Users/David/Desktop/kaazing-websocket-gateway-4.0.9/conf")
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    private JmxRule jmxConnection = new JmxRule(JMX_URI);
+
+    @Rule
+    public TestRule chain = RuleChain.outerRule(k3po).around(timeoutRule).around(gateway).around(jmxConnection);
+
+    @Specification("simple.echo.for.session.cnt")
+    @Test
+    @Ignore("https://github.com/kaazing/gateway/issues/331")
+    public void echoServiceCrossOriginAllow() throws Exception {
+        k3po.start();
+        k3po.awaitBarrier("SESSION_ESTABLISHED");
+        String attributeNames[] = {"NumberOfCurrentSessions", "NumberOfCurrentNativeSessions", "NumberOfCurrentEmulatedSessions",
+                "TotalMessageSentCount"};
+        // Could do this cleaner
+        MBeanServerConnection mbeanServerConn = jmxConnection.getConnection();
+        Set<ObjectName> mbeanNames = mbeanServerConn.queryNames(null, null);
+        String MBeanPrefix = "subtype=services,serviceType=echo";
+        for (ObjectName name : mbeanNames) {
+            if (name.toString().indexOf(MBeanPrefix) > 0) {
+                AttributeList attrValues = mbeanServerConn.getAttributes(name, attributeNames);
```

Comment:
If the test is **only** interested in `NumberOfCurrentSessions` attribute, then it's value can be obtained like this:

```java
     Object numberOfCurrentSessions = mbeanServerConn.getAttribute(name, "NumberOfCurrentSessions");
```